libs/template/config.go

```diff
@@ -125,12 +125,14 @@ func (c *config) skipPrompt(p jsonschema.Property, r *renderer) (bool, error) {
 		return false, nil
 	}
 
-	// Check if conditions specified by template author for skipping the prompt
-	// are satisfied. If they are not, we have to prompt for a user input.
-	for name, property := range p.Schema.SkipPromptIf.Properties {
-		if v, ok := c.values[name]; ok && v == property.Const {
-			continue
-		}
+	var keys []string
+	for k := range p.Schema.SkipPromptIf.Properties {
```

Comment:
We should also append all the fields in all schemas of `p.Schema.SkipPromptIf.AnyOf`. This went unnoticed because it's not covered by any test. Could you add a case for it? Basically:
```
	// Required values for skipping prompt not specified. Prompt should not be skipped.
	c.values["abc"] = "foobar"
	skip, err = c.skipPrompt(jsonschema.Property{
		Name:   "xyz",
		Schema: c.schema.Properties["xyz"],
	}, testRenderer())
	assert.NoError(t, err)
	assert.False(t, skip)
	assert.NotContains(t, c.values, "XYZ")
```

Please also add a one-line comment above that all fields referred to in a `skip_prompt_if` condition are implicitly made required and we diverge from strictly following the JSON schema because it makes the author UX better.

Reply:
Hmm we already had that test case actually and it passes, which makes sense right? If we make all the properties in all of the `anyOf` schemas required, then it would actually be wrong because they could specify only the properties of one of the schemas and that should be enough to pass.

Added the one-line comment though

Reply:
The tests were passing earlier because we were mutating the global `c.values` object. Values from preview test cases were not being cleaned up.